app/js/enhancedDate.js

```diff
@@ -1,5 +1,113 @@
 /* 
- * add your module definition for enhancedDate here
- * assign the module to a global variable
- * /
+ * This module will return an object with each of these methods defined on it. 
+ * This module should have a notion of "private" data, that is data that is used only internally by the module and is not exposed aspart of the public API.
+ * The module should handle the case when an API method (other than setDate) is called before a date has been set. 
+ * One approach to this is to check if a date has already been set, and if not, set a new date at that time. 
+ * Only the setDate method should be able to change the internally stored date after it has been set once.
+*/
+
+var enhancedDate = (function enhancedDate() {
+
+    'use strict';
+
+    //Private data, which is data that is used only internally by the module and is not exposed as part of the public API
+    var privateData = {
+        days : ['Sunday', 'Monday', 'Tuesday', 'Wednesday',
+                     'Thursday', 'Friday', 'Saturday'],
+        months : ['January', 'February', 'March', 'April', 'May',
+                   'June', 'July', 'August', 'September', 'October',
+                   'November', 'December']
+    };
+
+    // check if a date has already been set, and if not, set a new date at that time.
+    function checkHasDate(){
+    
+        if (!privateData.storedDate) {
+            privateData.storedDate = new Date();
+        }
+    }
+
+    var publicAPI = {
+
+        /*
+         * Stores the passed-in date. This function can take milliseconds after epoch or a Date object.  
+         * Do some type-checking to make sure you have a valid type. If no value is passed, default your internally stored date to now.
+        */
+        setDate: function setDate(inputDate) {
+
+            checkHasDate();
+
+            if (inputDate){
+
+                // check if a valid date type
+                if (Object.prototype.toString.call(inputDate) === '[object Date]' && !isNaN(inputDate)){
+                    privateData.storedDate = inputDate;
+                }
+
+                // check if pass milliseconds
+                if (isNaN(inputDate)){
+                    privateData.storedDate = new Date(inputDate);
+                }
+            }            
+        },
+
+        // Returns either milliseconds after epoch by default or a Date object if true is passed as an argument.        
+        getDate: function getDate(isDateObject) {
+            
+            checkHasDate();
+
+            return
+            {
```

Comment:
There's a bug here - always put your opening { on the same line as the code it comes after: return { 
What's happening here is the browser's ASI (automatic semi-colon insertion) is essentially doing this: return;
Additionally, you're trying to return either storedDate or storedDate.getTime( ), but you can't return it inside an object like this. Instead, you need to say return inside your if and your else blocks to return one or the other. You can also assign one or the other to a variable and return that variable to avoid using two separate return statements.